main.go

```diff
@@ -64,68 +47,36 @@ func applyVirtualMachineExport(vmNamespace, vmName string) error {
 		},
 	}
 
-	_, err = client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
+	_, err := client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
 	return err
 }
 
-func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
-	log.Printf("Downloading disk image from %s Virtual Machine...\n", vmName)
-
-	cmd := exec.Command("usr/bin/virtctl", "vmexport", "download", vmName, "--vm", vmName, "--volume", volumeName, "--output", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	if err := cmd.Run(); err != nil {
-		return err
-	}
-
-	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
-		return fmt.Errorf("File does not exist or is empty.")
-	}
-
-	log.Println("Download completed successfully.")
-	return nil
-}
-
-func decompressVirtualMachineDiskImage() error {
-	log.Println("Decompressing downloaded disk image...")
+func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName string) (string, error) {
+	log.Println("Getting raw disk URL from VirtualMachineExport object...")
 
-	diskImg, err := os.Open(diskPath)
-	if err != nil {
-		return err
+	vmExport, err := client.VirtualMachineExport(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
```

Comment:
Ensure that the URL is available right after VMExport is created (and not need to wait for it).

Reply:
is it possible that url is not available? If yes, isn't it cdi bug that it anounce something which is not ready?

Reply:
I'll check it. I see that it's in status and declared as a pointer, so we'd expect a nil returned instead URL.

Reply:
I would use something like for retry mechanism https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/modules/modify-data-object/pkg/dataobject/provider.go#L96